airflow/include/tasks/extract/astro_forum_docs.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from datetime import datetime
+
+import pandas as pd
+import pytz
+import requests
+from bs4 import BeautifulSoup
+from weaviate.util import generate_uuid5
+
+cutoff_date = datetime(2022, 1, 1, tzinfo=pytz.UTC)
+
+logger = logging.getLogger("airflow.task")
+
+
+def get_questions_urls(html_content: str):
+    soup = BeautifulSoup(html_content, "html.parser")
+    url_list = []
+    for a_tag in soup.findAll("a", class_="title raw-link raw-topic-link"):
+        url_list.append(a_tag.attrs.get("href"))
+    return url_list
+
+
+def get_publish_date(html_content):
+    soup = BeautifulSoup(html_content, "html.parser")
+    # TODO: use og:article:tag for tag filter
+    publish_date = soup.find("meta", property="article:published_time")["content"]
+    publish_date = datetime.fromisoformat(publish_date)
+    return publish_date
+
+
+def filter_cutoff_questions(questions_urls):
+    valid_urls = []
+    for question_url in questions_urls:
+        html_content = requests.get(question_url).content
+        if get_publish_date(html_content) < cutoff_date:
+            continue
+        valid_urls.append(question_url)
+    return valid_urls
+
+
+def get_cutoff_questions(forum_url):
+    page_number = 0
```

Comment:
can you add a doc string that this method is doing?

Reply:
Added